spec/requests/direct_messages_controller_spec.rb

```diff
@@ -116,5 +116,52 @@ def create_dm_channel(user_ids)
         post "/chat/direct_messages/create.json", params: { usernames: usernames }
       }.to change { UserChatChannelMembership.count }.by(3)
     end
+
+    context "when one of the users I am messaging has ignored, muted, or prevented DMs from the acting user creating the channel" do
+      let(:usernames) { [user1, user2, user3].map(&:username) }
+      let(:direct_message_user_ids) { [user.id, user1.id, user2.id, user3.id] }
+
+      shared_examples "creating dms with communication error" do
+        it "responds with a friendly error" do
+          expect {
+            post "/chat/direct_messages/create.json", params: { usernames: [usernames] }
+          }.not_to change { DirectMessageChannel.count }
+          expect(response.status).to eq(422)
+          expect(response.parsed_body["errors"]).to eq([I18n.t("chat.errors.not_accepting_dms", username: user1.username)])
+        end
+      end
+
+      describe "user ignoring the actor" do
+        before do
+          Fabricate(:ignored_user, user: user1, ignored_user: user, expiring_at: 1.day.from_now)
```

Comment:
does it need to be in a before block? (same question for others after

Reply:
Yes since they are all testing different ways the "creating dms with communication error" shared example could fail